From: Colomban Wendling <cwendling@hypra.fr>
Date: Tue, 25 Mar 2025 11:00:50 +0100
Subject: Fix libsystemd auto-detection
Origin: upstream, https://github.com/mate-desktop/mate-control-center/commit/b5f6b8cf2d1976e34c20d37ab737545713174ede

Properly handle missing libsystemd when enable-systemd=auto, which is
even the default for this switch.

Without this, one had to explicitly give `--disable-systemd` despite
what the help text stated.
---
 configure.ac | 15 +++++----------
 1 file changed, 5 insertions(+), 10 deletions(-)

diff --git a/configure.ac b/configure.ac
index 4aa6af2..7ecadad 100644
--- a/configure.ac
+++ b/configure.ac
@@ -179,16 +179,11 @@ AS_IF([test "x$enable_appindicator" = "xyes"], [
 
 have_systemd=no
 AC_ARG_ENABLE(systemd, AS_HELP_STRING([--disable-systemd], [disable systemd support]),enable_systemd="$enableval",enable_systemd=auto)
-if test "x$enable_systemd" != "xno"; then
-	PKG_CHECK_MODULES(SYSTEMD, [libsystemd], [have_systemd=yes],
-	                  [PKG_CHECK_MODULES(SYSTEMD, [libsystemd >= $SYSTEMD_REQUIRED],
-	                  [have_systemd=yes])])
-	if test "x$have_systemd" = xno && test "x$enable_systemd" = xyes; then
-		AC_MSG_ERROR([*** systemd support requested but libraries not found])
-	elif test "x$have_systemd" = xyes; then
-		AC_DEFINE(HAVE_SYSTEMD, 1, [Define if systemd is available])
-	fi
-fi
+AS_CASE(["$enable_systemd"],
+        [yes], [PKG_CHECK_MODULES(SYSTEMD, [libsystemd >= $SYSTEMD_REQUIRED], [have_systemd=yes])],
+        [auto], [PKG_CHECK_MODULES(SYSTEMD, [libsystemd >= $SYSTEMD_REQUIRED], [have_systemd=yes], [have_systemd=no])],
+        [have_systemd=no])
+AS_IF([test "x$have_systemd" = xyes], [AC_DEFINE(HAVE_SYSTEMD, 1, [Define if systemd is available])])
 
 AM_CONDITIONAL(HAVE_SYSTEMD, [test "$have_systemd" = "yes"])
 
-- 
2.47.3
